server2/Arcadia.Assistant/Arcadia.Assistant.Vacations/Notification/VacationsNotification.cs

```diff
@@ -0,0 +1,270 @@
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Autofac.Features.OwnedInstances;
+
+    using Contracts;
+
+    using Employees.Contracts;
+
+    using Microsoft.Extensions.Logging;
+
+    using Notifications.Contracts;
+    using Notifications.Contracts.Models;
+
+    using NotificationTemplates;
+    using NotificationTemplates.Configuration;
+
+    using Organization.Contracts;
+
+    public class VacationsNotification
+    {
+        private readonly IEmployees employeeService;
+        private readonly ILogger logger;
+        private readonly Dictionary<string, INotificationConfiguration> notificationConfigurations;
+        private readonly INotifications notifications;
+        private readonly IOrganization organizationService;
+        private readonly Func<Owned<VacationsStorage>> storageFactory;
+
+        public VacationsNotification(
+            Func<Owned<VacationsStorage>> storageFactory,
+            INotifications notifications,
+            IEmployees employeeService,
+            IOrganization organizationService,
+            IVacationsStatusChangeNotificationConfiguration vacationsCreateNotificationConfiguration,
+            IVacationsApproveRequireNotificationConfiguration vacationsApproveRequireNotificationConfiguration,
+            ILogger<VacationsNotification> logger)
+        {
+            this.notifications = notifications;
+            this.storageFactory = storageFactory;
+            this.employeeService = employeeService;
+            this.organizationService = organizationService;
+            this.notificationConfigurations = new Dictionary<string, INotificationConfiguration>
+            {
+                { VacationsNotificationTemplate.VacationsStatusChanged, vacationsCreateNotificationConfiguration },
+                {
+                    VacationsNotificationTemplate.VacationsApproveRequire,
+                    vacationsApproveRequireNotificationConfiguration
+                }
+            };
+            this.logger = logger;
+        }
+
+        public async Task SendVacationCreateNotification(
+            EmployeeId employeeId, VacationDescription vacationDescription, CancellationToken cancellationToken)
+        {
+            this.logger.LogDebug(
+                "Send new employee (id={EmployeeId}) vacation request notification for dates {StartDate}-{EndDate}.",
+                employeeId, vacationDescription.StartDate, vacationDescription.EndDate);
+
+            var status = "Created";
+            var owner = await this.GetEmployeeMetadataAsync(employeeId, cancellationToken);
+            if (owner == null)
+            {
+                this.logger.LogError("Can't find owner employee.");
+                throw new ArgumentException("Owner employee missed in database.");
+            }
+
+            await this.SendStatusChangedNotification(
+                owner!,
```

Comment:
is exclamation mark needed? Compiler should not generate warnings because there is a check above. Please check and remove it if compiler doesn't complain.